### DISCUSSION TITLE
No field matches populate when importing a csv

### DISCUSSION THREAD

**Discussion by rguttersohn** [konnco/filament-import#124] [Q&A] [answered]
I am sure I've missed a step somewhere. When I drag and drop the csv file, it uploads correctly, but the corresponding fields do not populate with the fields from the csv file. 

I am using:

- filament:3.0-stable
- filament-import:2.0.1-beta
- Laravel:10x

For this example, I have  a table called Posts with three columns: id (auto increment), title, author.

Here is my ListPosts.php file:

```php

namespace App\Filament\Resources\PostResource\Pages;

use App\Filament\Resources\PostResource;
use Filament\Actions;
use Filament\Resources\Pages\ListRecords;
use Konnco\FilamentImport\Actions\ImportAction;
use Konnco\FilamentImport\ImportField;

class ListPosts extends ListRecords
{
    protected static string $resource = PostResource::class;

    protected function getHeaderActions(): array
    {
        return [
            ImportAction::make()
                ->fields([
                    ImportField::make('title')
                        ->required(),
                    ImportField::make('author')
                        ->required(),
                ])
            ];
    }

  
}

```

When I manually update the table using the form, the data is saved accordingly. 

The closest issue I found was [here](https://github.com/konnco/filament-import/issues/122) where a user was getting an error that I also received when I tried to see if uploading an xlsx file would work. Not sure if the issues are related. 

Here is my config file:

```php
return [
    'accepted_mimes' => [
        'application/vnd.ms-excel',
        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
        'text/csv',
        'text/plain',
        'csv',
        'txt',
        'application/csv'
    ],
    'temporary_files' => [
        'disk' => 'local',
        'directory' => 'filament-import',
    ],
];

```

**Comment by rguttersohn** [comment]
More details:

I've confirmed the csv files are saving under store/app/filament-import. 

Here is what I see after uploading a file:

<img width="1068" alt="Screen Shot 2023-08-31 at 11 53 10 PM" src="https://github.com/konnco/filament-import/assets/54447208/58c23197-c970-4dfb-a520-941a5536357d">

**Reply by tiagof** [reply]
Can you show us the file content?

**Reply by rguttersohn** [reply]
Here is the csv file (It was made using excel but saved as a csv if that makes a difference with encoding for some reason):
[posts_import.csv](https://github.com/konnco/filament-import/files/12503807/posts_import.csv)

Here is my form schema:

```php
public static function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('title'),
                TextInput::make('author')
            ]);
    }
```

**Reply by tiagof** [reply]
But why are you using TextInput? You should use ImportField according to the documentation..

Edit: apologies if my request was not clear  I meant to ask for the ImportAction schema, not the form schema.

**Reply by rguttersohn** [reply]
Ah gotcha. Yeah that’s my form schema. My importaction schema is pasted above in my initial post.

**Reply by tiagof** [reply]
Ah, yes, sorry I missed it. 
It is strange. I'm using 2.0.0- beta and I'm importing csv files fine. 
Can you try downgrading tot his version?

Tomorrow I should have a bit more time and will try to use your csv to see if I can replicate the issue.

**Reply by rguttersohn** [reply]
yeah I am going to do more investigating. I am also using docker to host my local dev site. Maybe it's a permissions issue.

**Reply by rguttersohn** [reply]
After some investigating, I found that the files are being uploaded to storage/framework/caches/laravel-excel and sometimes storage/app/filament-import and storage/app/livewire-tmp. 

The files in /laravel-excel are -- for some reason -- saving as blank txt files. Could this be the issue? I cannot figure out why they are being saved as txt AND are blank. 

Or are the files suppose to be in /filament-import or /livewire-tmp? Which directory do I need to pay attention to when it comes to filament-import?

**Reply by adriaardila** [reply]
> After some investigating, I found that the files are being uploaded to storage/framework/caches/laravel-excel and sometimes storage/app/filament-import and storage/app/livewire-tmp. 
> 
> The files in /laravel-excel are -- for some reason -- saving as blank txt files. Could this be the issue? I cannot figure out why they are being saved as txt AND are blank. 
> 
> Or are the files suppose to be in /filament-import or /livewire-tmp? Which directory do I need to pay attention to when it comes to filament-import? 

Can you test if the 2.0.0-beta version works for you? If it does it means the PR i did fixing remote disk uploads like S3 broke something for local uploads so it should be an easy fix.

**Reply by rguttersohn** [reply]
I switched to 2.0.0-beta and it's still not working. 

I also created a new table and new filament resource for the table and now I am getting an error saying "Maatwebsite \ Excel \ Exceptions \ NoTypeDetectedException".' I am new to Laravel and am having a hard time reading the errors to figure out where it's coming from. 

I deleted all of the temporary files from /filament-import, /livewire-tmp and framework/cache/laravel-excel to see what happens when I attempt an import. They are saved under filament-import AND laravel-excel. However, the laravel-excel files are convert from csv to txt files and are blank. The ones saved in filament-import look good to me. 

I wonder if my issue is more of a config issue than something up with the package version. I just don't know where to turn to address this. Could using docker be causing the issue?

**Reply by rguttersohn** [reply]
OK didn't see the dropdown for vendor frames. Here is where the exception is being thrown from vendor / maatwebsite / excel / src / Factories / ReaderFactory.php : 84

```php
private static function identify(TemporaryFile $temporaryFile): string

    {

        try {

            return IOFactory::identify($temporaryFile->getLocalPath());

        } catch (Exception $e) {

            throw new NoTypeDetectedException(null, null, $e);

        }

    }

```

That to me reads like its a config issue and that php is not obtaining the localpath to my temporary file.

**Reply by adriaardila** [reply]
Okay, the thing you mention about the laravel-excel files being blanks happened to me too and in my case it was because the library was looking for the livewire-tmp folder in my local storage, even when I had it setup to s3. Did you change anything in your livewire.php config?

Try setting it like this:

```
'temporary_file_upload' => [
        'disk' => 'local',        // Example: 'local', 's3'              | Default: 'default'
        'rules' => null,       // Example: ['file', 'mimes:png,jpg']  | Default: ['required', 'file', 'max:12288'] (12MB)
        'directory' => 'livewire-tmp',   // Example: 'tmp'                      | Default: 'livewire-tmp'
        'middleware' => null,  // Example: 'throttle:5,1'             | Default: 'throttle:60,1'
        'preview_mimes' => [   // Supported file types for temporary pre-signed file URLs...
            'png', 'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpg', 'jpeg', 'mpga', 'webp', 'wma', 'ttf'
        ],
        'max_upload_time' => 5, // Max duration (in minutes) before an upload is invalidated...
    ],
```

Then, when you upload the xlsx or csv, make sure it's actually uploaded in the livewire-tmp local folder, if it is not there then the importer won't find the file since it looks for the tmp file stored by livewire.

Also check that permissions in the folder are correct and files can be written to it

**Reply by adriaardila** [reply]
I just tested it and indeed I broke something with the last PR, let me fix it and make a PR

**Comment by adriaardila** [comment, marked as answer]
Just made a PR to fix this issue with local disks, apparently my last PR fixing s3 broke not remote disks.

Tested it with local disk as well as s3 disk and all tests passing, so it should be fine now

https://github.com/konnco/filament-import/pull/125

**Reply by rguttersohn** [reply]
Great. Your PR fixed my issue.
